sw/device/tests/keymgr_dpe_key_derivation_test.c

```diff
@@ -42,64 +42,187 @@ static void init_kmac_for_keymgr(void) {
+              dif_keymgr_dpe_generate_params_t *params) {
+  CHECK_DIF_OK(dif_keymgr_dpe_generate(keymgr_dpe, params));
+  CHECK_STATUS_OK(keymgr_dpe_testutils_wait_for_operation_done(keymgr_dpe));
+}
+
 bool test_main(void) {
-  // Initialize keymgr_dpe
   dif_keymgr_dpe_t keymgr_dpe;
-  dif_keymgr_dpe_advance_params_t keymgr_dpe_advance_params;
-
-  CHECK_STATUS_OK(keymgr_dpe_testutils_startup(&keymgr_dpe));
+  dif_keymgr_dpe_advance_params_t adv_params;
+  dif_keymgr_dpe_generate_params_t gen_params;
 
+  // Start keymgr_dpe, letting it derive the boot stage 0 key into slot 1.
+  CHECK_STATUS_OK(keymgr_dpe_testutils_startup(&keymgr_dpe,
+                                               /*slot_dst_sel=*/1));
   CHECK_STATUS_OK(keymgr_dpe_testutils_check_state(
       &keymgr_dpe, kDifKeymgrDpeStateAvailable));
-  LOG_INFO("KeymgrDpe state available");
-
-  keymgr_dpe_advance_params.slot_src_sel = 1;
-  keymgr_dpe_advance_params.slot_dst_sel = 1;
-  keymgr_dpe_advance_params.max_key_version = 0x100;
-  keymgr_dpe_advance_params.binding_value[0] = 0xd;
-  keymgr_dpe_advance_params.binding_value[1] = 0xe;
-  keymgr_dpe_advance_params.binding_value[2] = 0xa;
-  keymgr_dpe_advance_params.binding_value[3] = 0xd;
-  keymgr_dpe_advance_params.binding_value[4] = 0xb;
-  keymgr_dpe_advance_params.binding_value[5] = 0xe;
-  keymgr_dpe_advance_params.binding_value[6] = 0xe;
-  keymgr_dpe_advance_params.binding_value[7] = 0xf;
-  keymgr_dpe_advance_params.slot_policy = 0x5;
+  LOG_INFO("KeymgrDpe derived boot stage 0 key");
 
   init_kmac_for_keymgr();
 
-  CHECK_STATUS_OK(keymgr_dpe_testutils_advance_state(
-      &keymgr_dpe, &keymgr_dpe_advance_params));
-  CHECK_STATUS_OK(keymgr_dpe_testutils_check_state(
-      &keymgr_dpe, kDifKeymgrDpeStateAvailable));
-  LOG_INFO("KeymgrDpe updated UDS slot");
-
-  CHECK_STATUS_OK(keymgr_dpe_testutils_wait_for_operation_done(&keymgr_dpe));
-
-  dif_keymgr_dpe_generate_params_t genparams;
-  genparams.key_dest = kDifKeymgrDpeKeyDestAes;
-  genparams.sideload_key = false;
-
-  genparams.salt[0] = 0xd;
-  genparams.salt[1] = 0xe;
-  genparams.salt[2] = 0xa;
-  genparams.salt[3] = 0xd;
-  genparams.salt[4] = 0xb;
-  genparams.salt[5] = 0xe;
-  genparams.salt[6] = 0xe;
-  genparams.salt[7] = 0xf;
-  genparams.version = 0;
-  genparams.slot_src_sel = 1;
+  // // Generate HW output to AES from boot stage 0 key.
+  // gen_params.slot_src_sel = 1;
+  // gen_params.sideload_key = true;  // HW key
+  // gen_params.key_dest = kDifKeymgrDpeKeyDestAes;
+  // gen_params.version = 0;
+  // gen_params.salt[7] = 0x72d5886b;
+  // gen_params.salt[6] = 0x4e359e52;
+  // gen_params.salt[5] = 0x0d7ff336;
+  // gen_params.salt[4] = 0x267773cf;
+  // gen_params.salt[3] = 0x00c7d10c;
+  // gen_params.salt[2] = 0x6dea4fb9;
+  // gen_params.salt[1] = 0x77fa328a;
+  // gen_params.salt[0] = 0x15779805;
+  // generate(&keymgr_dpe, &gen_params);
+  // LOG_INFO("KeymgrDpe generated AES output from boot stage 0");
```

Comment:
why is this block commented out?

Reply:
I was having some problems with this earlier, but now it works, so I re-enabled it. I'll squash the commits and get this ready for review